### DISCUSSION TITLE
Diatone Mamba Basic F722 Mini MK3 - Magnetometer not working in INAV

### DISCUSSION THREAD

**Discussion by Ram-85** [iNavFlight/inav#6781] [Q&A] [answered]
Hi,

I am using this board for my Micro LR build with the Matek M8Q-5883 GPS/Compass. Everything seems to work except for the Magnetometer (red). In Diatone's website they have the following command to set the Magnetometer, but they dont seem to work with iNav:
set mag_i2c_device = 1
set mag_bustype = i2c

I am assuming the issue is with SCL SDA and I2C mapping in INAV, Is there an alternative to these commands in iNav?

Here is the wiring diagram for your reference:
https://cdn.shopify.com/s/files/1/0027/2708/4144/files/4-MAMBA_Basic_F722_MINI_MK3.pdf?v=1610714258

Thanks

**Comment by JulioCesarMatias** [comment]
What is the model of the selected compass? In configurations >> Sensors >> Magnetometer
![Captura de tela 2021-04-03 155059](https://user-images.githubusercontent.com/25625982/113488435-83120900-9494-11eb-9a6d-15e6a54a0a55.png)

**Reply by Ram-85** [reply]
It is QMC5883. This is not my first build with the Matek M8Q-5883 GPS/Compass, I have other builds with different FC and it works fine in them.

**Reply by JulioCesarMatias** [reply]
> É QMC5883. Esta não é minha primeira compilação com o Matek M8Q-5883 GPS / Compass, tenho outras compilações com FC diferentes e funciona bem nelas.

This symbol is true for you?
![Captura de tela 2021-04-03 175052](https://user-images.githubusercontent.com/25625982/113491175-2ec35500-94a5-11eb-876e-93ed7cac633f.png)

**Reply by Ram-85** [reply]
No it is red!
![image](https://user-images.githubusercontent.com/80753711/113496201-d5ffb680-94bc-11eb-82c6-e1aba16a2501.png)

**Reply by JulioCesarMatias** [reply]
> No it is red!
> ![image](https://user-images.githubusercontent.com/80753711/113496201-d5ffb680-94bc-11eb-82c6-e1aba16a2501.png)

try with scl and sda inverted.

**Reply by JulioCesarMatias** [reply]
Do you have a multimeter to measure the voltage supply to the compass? It should be 5v, and after the 3.3v voltage regulator

**Reply by Ram-85** [reply]
I checked the voltage, and it is fine. I tried to flash Betflight on it, and the compass worked fine. I guess it is an issue with INAV firmware for this board.
![image](https://user-images.githubusercontent.com/80753711/113496692-d484bd00-94c1-11eb-8473-d901d664f10d.png)

**Reply by JulioCesarMatias** [reply]
> Verifiquei a voltagem e está tudo bem. Tentei fazer um flash Betflight nele, e a bússola funcionou bem. Eu acho que é um problema com o firmware INAV para esta placa.
> ![imagem](https://user-images.githubusercontent.com/80753711/113496692-d484bd00-94c1-11eb-8473-d901d664f10d.png)

when you wrote `set mag_i2c_device = 1`  and ` set mag_bustype = i2c`  in the iNav CLI did you wrote `save` at the end?

```
set mag_hardware = AUTO
set mag_bustype = I2C
set mag_i2c_device = 1
save
```

restart the FC after that

**Reply by Ram-85** [reply]
It gives me an error
![image](https://user-images.githubusercontent.com/80753711/113497227-8ffc2000-94c7-11eb-9688-6bbceb89f39a.png)

**Reply by Ram-85** [reply]
I also tried these commands that were suggested by Diatone Support:
![image](https://user-images.githubusercontent.com/80753711/113497257-df425080-94c7-11eb-92bf-9ccadd0413e5.png)

**Reply by JulioCesarMatias** [reply]
the commands don't really exist >> https://github.com/iNavFlight/inav/blob/master/docs/Settings.md

try to use the `auto`  type in magnetometer in the settings

unfortunately I can't help anymore, probably a problem with the iNAV firmware

sorry for my bad english, it's not my language

**Reply by Ram-85** [reply]
I already tried the Auto settings and did not work. 
Thanks for trying to help, I appreciate it!

**Reply by julianschweizer** [reply]
I had to set it to "AUTO" and then back to "HMC***" and powerloop repeatedly to get magnetometer working (Mamba F405 MK2 & BN-880Q). Just give it a try ... .

**Comment by Jetrell** [comment]
@Ram-85 It appears that the Mamba F722 has the I2C bus defined as default Mag/Compass, shared with the Barometer and broken out on TX3_RX3.
https://github.com/iNavFlight/inav/blob/master/src/main/target/MAMBAF722/target.h

**Comment by Ram-85** [comment, marked as answer]
It seems this board is not supported at this time according to **robertha64** 's comment below
https://github.com/iNavFlight/inav/issues/6774#issuecomment-819670382
